src/rdycore.c

```diff
@@ -18,9 +18,6 @@ PetscErrorCode RDyInit(int argc, char *argv[], const char *help) {
       PetscCall(PetscInitialize(&argc, &argv, (char *)0, (char *)help));
     }
 
-    // some exit statuses don't get propagated without this vvv
-    PetscCall(PetscPushErrorHandler(PetscMPIAbortErrorHandler, NULL));
```

Comment:
Do we not need this anymore? Does this error handler cause problems with AMR?

Reply:
We didn't have the corresponding `PetscPopErrorHandler` in the code. Thus, when running with `-malloc_debug`, we got errors. Removing this fixed the error reporting. I don't know if adding `PetscPopErrorHandler` is a better solution.

Reply:
Let's see how things work without it and revisit if needed.